src/sagemaker_xgboost_container/data_utils.py

```diff
@@ -288,7 +341,107 @@ def get_libsvm_dmatrix(files_path):
     return dmatrix
 
 
-def get_dmatrix(data_path, content_type, csv_weights=0):
+def get_parquet_dmatrix(files_path):
+    try:
+        table = pq.read_table(files_path)
+
+        data = table.to_pandas()
```

Comment:
Just a warning, but pandas is really slow. Can we avoid using pandas here?

Reply:
Unfortunately this seems to be the only method of converting the pyarrow Table into a format that can be read by DMatrix. This is reflected by file mode training being slower than pipe mode. I'm not certain if there's a way to verify and enforce the conversion, but I will look into it.